### PR TITLE
Add infinite scrolling prompt generation

### LINKED ISSUE #13: Add infinite prompt generation
We want the user to be able to receive an infinite stream of varying words to type, with the ending conditions being the time limit running out or if the user stops the test.

We need to find a way to generate varied words from a source and use that as the target string which will get added to as the user progresses.

### PR DESCRIPTION
**Pull request by andrewjleung**
# What

- Use the `babble` package to infinitely generate words for the target string.
- Modify the view to display a page consisting of the last x lines of y target words (x and y are configurable), scrolling as the user completes lines.

## Demo

stronglytyped-prompt-gen-demo

# Why

A time-based typing test should never run out of words before the time runs out. It should also test the user with a wide variety of words.

Additionally, the user should always be able to see some number of words ahead of the word that they are typing so that they may read ahead while typing. Having words to type shouldn't be a limiting factor in the speed at which they type.

# Notes

- Generating words is possibly costly since for each generation, we need to filter ALL the words used by `babble` by the configured max length. This will be addressed in a follow-up adding another word source.
- This also addresses the edge case of deleting the first word in a page. The behavior is that the user can only scroll forwards, not backwards. They cannot delete past the first word of the current page, even if there is a previous word which is not correct.

### CODE REVIEW

**Comment by kartik-s** [comment]
The `babble` docs mention this, but in case anyone hits it: you need `/usr/share/dict/words` present on your system o/w you get a panic

**Comment by kartik-s** [comment]
the paging is so 🤌

File: src/utils.go [hunk 1]
```diff
@@ -6,3 +6,19 @@ func min(a, b int) int {
 	}
 	return a
 }
+
+func filter[T any](list []T, test func(T) bool) (ret []T) {
```
Comment (kartik-s) [inline src/utils.go:10]: nit: I think this is unused?

**Review by kartik-s — APPROVED** [review]
Nicely done :100:

File: src/utils.go (hunk 1 above)
Comment (andrewjleung) [reply src/utils.go:10]: Yeah, I think it won't hurt to keep / I do use it in https://github.com/ideapeople/stronglytyped/pull/17 so I think it's okay to keep

Comment (kartik-s) [reply src/utils.go:10]: Ah didn't see that!

Comment (andrewjleung) [reply src/utils.go:10]: How dare you!!!!!!!!!! No worries tho mate I've kept that PR a secret 🤫

**Comment by andrewjleung** [comment]
> The babble docs mention this, but in case anyone hits it: you need /usr/share/dict/words present on your system o/w you get a panic

@kartik-s Yeah I'm not a huge fan of the dependency on the unix words being present, it's also funny because [babble on windows just defines an array with all the dictionary words](https://raw.githubusercontent.com/tjarratt/babble/master/babble_windows.go) 🥲

Maybe we can pivot away from it in the future or copy that list of words, https://github.com/ideapeople/stronglytyped/pull/17 adds a common words file too that will probably end up being the default words we use.

**Pull request merged**